lib/core/src/Cardano/Wallet/DB/Sqlite/Types.hs

```diff
@@ -657,3 +662,27 @@ instance PersistField DerivationPrefix where
 
 instance PersistFieldSql DerivationPrefix where
     sqlType _ = sqlType (Proxy @Text)
+----------------------------------------------------------------------------
+-- Other
+
+instance PersistField POSIXTime where
+    toPersistValue = PersistText
+        . T.pack
+        . formatTime defaultTimeLocale (iso8601DateFormat (Just "%H:%M:%S"))
+        . posixSecondsToUTCTime
+    fromPersistValue (PersistText time) =
+        utcTimeToPOSIXSeconds <$>
+            parseTimeM True defaultTimeLocale
+                (iso8601DateFormat (Just "%H:%M:%S")) (T.unpack time)
+    fromPersistValue _ = Left
+        "Could not parse POSIX time value"
```

Comment:
**Question**: is there a round-trip test for the textual serialization/deserialization of `POSIXTime`?

If not, I would suggest adding such a test.

I would recommend the following:

1. Move the textual serialization/deserialization logic to a pair of `ToText` and `FromText` instances.
2. Write a `textRoundtrip` test in `Cardano.Wallet.DB.Sqlite.TypesSpec`.
3. Re-express the `PersistField` instances in terms of your `ToText` and `FromText` instances.

Reply:
`ToText` and `FromText` are bad classes imo. These do all kinds of serialisation/deserialisation from and to unknown formats with very weak type-safety. I've been bitten by them before and I don't like the approach.

Reply:
> `ToText` and `FromText` are bad classes imo. These do all kinds of serialisation/deserialisation from and to unknown formats with very weak type-safety. I've been bitten by them before and I don't like the approach.

I sympathize with your concerns. However, `ToText` and `FromText` are currently our standard tools for handling textual serialization/deserialization. We could change this, but I'm sure you'll agree that's outside the scope of this PR.

As for type-safety, we take care to define context-wrappers where necessary (for example, `ApiT`), to allow context-dependent serializations.

> Also: the serialisation/deserialisation is already covered by the database test: https://github.com/input-output-hk/cardano-wallet/pull/2249/files#diff-c3e090fb717915e32577d1c7311e8ab5392c830dec9a705e025f0bf84af6b099R1443

I don't think we should rely on this property alone to indirectly test text serialization. It's important that we also test bad inputs, for example. For examples, see: https://github.com/input-output-hk/cardano-wallet/blob/368b9fc27249e1faf6577c020297b74d4cf6f161/lib/core/test/unit/Cardano/Wallet/Primitive/TypesSpec.hs#L776

Reply:
> I sympathize with your concerns. However, ToText and FromText are currently our standard tools for handling textual serialization/deserialization. We could change this, but I'm sure you'll agree that's outside the scope of this PR.

Sorry, I'd prefer not to use these classes, unless there's a really strong case for them. I don't see it.

> I don't think we should rely on this property alone to indirectly test text serialization. It's important that we also test bad inputs, for example.

Then we need a proper unit test for the Persist classes, which is what's really to be tested here.

Reply:
@jonathanknowles wrote:
>> `ToText` and `FromText` are currently our standard tools for handling textual serialization/deserialization.

@hasufell wrote:
> Sorry, I'd prefer not to use these classes, unless there's a really strong case for them. I don't see it.

The benefits of using these classes:

- Consistency with the rest of the code. We use these classes pervasively for textual serialization & deserialization.
- Ability to reuse the existing `textRoundtrip` testing infrastructure.

If you want to avoid writing `ToText` instances, then as an alternative you could provide a test using the `persistRoundtrip` function.